.github/workflows/compatibility.yml

```diff
@@ -36,40 +37,47 @@ jobs:
     strategy:
       fail-fast:    false
       max-parallel: 10
-      matrix: ${{ fromJSON(needs.generate-matrix.outputs.matrix) }}
+      matrix: ${{ fromJSON(needs.generate-first-matrix.outputs.matrix) }}
     steps:
       # clone the repository
       - uses: actions/checkout@v3
       # enable dependencies caching
       - uses: actions/cache@v3
         with:
           path: ~/.cache/composer/
-          key:  ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('composer.lock') }}
       # setup PHP, but without debug extensions for reasonable performance
       - uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
-          tools:       composer
-          coverage:    none
-      # No longer needed. Early version of PHPUnit is already part of repository.
-      # run: if [[ "${{ matrix.php }}" == '7.0' ]]; then wget https://phar.phpunit.de/phpunit-6.5.14.phar && mv phpunit-6.5.14.phar phpunit.phar; fi;
+          tools: composer
+          coverage: none
       # run CI checks
       - run: bash bin/run-ci-tests.bash
+  
+  generate-second-matrix:
+    name: "Generate the matrix for compatibility-woocommerce-beta dynamically"
+    runs-on: ubuntu-20.04
+    outputs:
+      matrix: ${{ steps.generate_matrix.outputs.matrix }}
+    steps:
+      - name: "Generate matrix"
+        id: generate_matrix
+        run: |
+          PHP_VERSIONS=$( echo "[\"$PHP_MIN_VERSION\", \"8.0\", \"8.1\"]" )
```

Comment:
Will this require a manual change once we bump this further? Should we create the subsequent versions also dynamically or is it intended to be bumped like this to avoid breaking changes?

Reply:
I don't have enough history here, I'm not sure how we previously bumped the "max versions" for PHP.
I guess there would be concerns about breaking changes and compatibility for sure, so it doesn't seem wise to bump it at the same time as we bump the min version.

We can plan another task to allow bumping the `max` and `max - 1` PHP versions based on a new parameter `php_max_version`, but doesn't seem that urgent